pywisp/gui.py

```diff
@@ -1244,6 +1244,10 @@ def closeEvent(self, QCloseEvent):
         """
         if self.isConnected:
             self.disconnect()
+        if self.exporter != None:
```

Comment:
Instead of ignoring the close event, it would be better to wait for the exporters worker here (thus moving the logic from Exporter.__del__ here) since the timing of the __del__ call seems to be non deterministic. But the handling of the close event is.

I will add a test case that showcases the issue.

Reply:
I think this is a simple (and always correct) way of dealing with this. The user is notified, so knows what to expect, and can just try closing the window later.

Before implementing this, I had tried actually waiting and then continuing to shut down, but none of the attempts actually worked:
- waiting on the thread would just block without ever actually shutting down
- using a timer that would just call the close event again at a later time would raise errors about using things in python that were deleted on the c++ side

but if you work out  a way that works, I'm happy to reconsider

Reply:
You are right, this way definitively works. Furthermore I did not know that you already invested so much time into the issue.

As shown by the test case, overloading the `__del__` call does in fact not stop the deletion, so I moved the logic there into the `closeEvent` handler on friday and on the two test systems that I have available it worked as expected. That was actually the reason for my initial remark on this.
As for why this approach did not work on your machine I have no idea and before taking the chance of the same thing happening elsewhere I think staying with the solution you provided is the best way to go forward.

I think we can safely remove the  `__del__`  handler, though.

Reply:
huh, don't remember what I must have been doing wrong, last week, but moving the waiting for a thread into a method and calling it normally actually works...

fix incoming

Reply:
I like it. Having Exporter.wait() saves outsiders from directly interacting with the internal worker Thread.